packages/client/src/views/SearchResult/AdvancedSearch.tsx

```diff
@@ -85,16 +77,33 @@ export const isAdvancedSearchFormValid = (value: IBaseAdvancedSearchState) => {
       Boolean(value[key as keyof IBaseAdvancedSearchState])
   )
   //handle date fields separately
-  const validDateFields = dateFieldTypes.filter(
-    (key) =>
-      value[key as keyof IBaseAdvancedSearchState] &&
-      isValidDateRangePickerValue(
-        value[key as keyof IBaseAdvancedSearchState] as IDateRangePickerValue
-      )
+  const validationResultsPerDateField = dateFieldTypes.map((key) => {
+    const dateObj = value[
+      key as keyof IBaseAdvancedSearchState
+    ] as IDateRangePickerValue
+
+    if (!dateObj) {
+      return false
+    }
+
+    if (!isValidDateRangePickerValue(dateObj)) {
+      return false
+    }
+
+    if (dateObj.exact && isInvalidDate(dateObj.exact)) {
```

Comment:
Is it intentional that dates without leading zeros don't work?
<img width="639" alt="image" src="https://github.com/opencrvs/opencrvs-core/assets/1206987/a4f8d12f-ec94-4216-94da-de9b2345b41c">

Reply:
Would something like this work?

```ts
import { isValid, parse } from 'date-fns';

const validateDate = (dateString) => {
  const formats = ['yyyy-M-d', 'yyyy-MM-dd'];
  return formats.some(format => isValid(parse(dateString, format, new Date())));
};

console.log(validateDate('2020-2-3'));  // true
console.log(validateDate('2020-02-03')); // true
console.log(validateDate('2020-2-30'));  // false
console.log(validateDate('2020-02-30')); // false
```

Reply:
The best option is that the date input component would always output the given date in the same format

Reply:
Yeah I was trying to see how I could make it work with the current comparison of the converted iso string, but i guess this is a more simplified approach.